extensions/pkg/controller/worker/genericactuator/actuator_restore.go

```diff
@@ -138,8 +138,12 @@ func (a *genericActuator) restoreMachineSetsAndMachines(ctx context.Context, log
 				}
 			}
 
-			newMachine.Status = machine.Status
-			return a.client.Status().Update(ctx, newMachine)
+			// Patch() is used here instead of Update() so that only the machine.Status.Node field is modified as a workaround
```

Comment:
Does it make sense to add a todo so we don't forget to switch back to update?

```suggestion
			// TODO (plkokanov): Switch back to Update() when https://github.com/gardener/machine-controller-manager/issues/642 is resolved.
			// Patch() is used here instead of Update() so that only the machine.Status.Node field is modified as a workaround
```

Reply:
I was wondering whether to add such a comment, but decided to leave it out since `Patch()` works just as well. 
Btw there is currently a discussion with the MCM maintainers whether we can stop using the `status.Node` field and instead rely on the `node` label in the machine objects. So in the future I would prefer to completely remove the status modification if possible. I can actually open an issue so that we track such a possibility for a change.

Reply:
@timebertt is there a particular reason we would like to switch to Update() in the future, or is the Patch() in this usage does the same job?

Reply:
IIRC, I changed it to `Update` because the code was modifying the entire status section, which was not safe to be done without optimistic locking.
However, it seems that we only want to modify one specific field (`machine.Status.Node`), which is safe to do without optimistic locking.
Hence, it should not be needed to switch to `Update()` again in the future.